### PR TITLE
Add option for custom keybinds for navigation

### LINKED ISSUE #211: Set keybindings for up/down... movement
I don't notice a way to set my own bindings for scroll up, down (and probably some other basic movements such as left, right too).

Sorry if I'm missing a way to set these, but it seems they must be hardcoded somewhere to arrow keys and hjkl. 

Optionally setting these keys would be helpful for users with alternative keyboard layouts, or users may just have some other preference for how to move about.

### PR DESCRIPTION
**Pull request by singalhimanshu**
Fixes #211

### CODE REVIEW

File: display/help.go [hunk 1]
```diff
@@ -78,6 +78,10 @@ func helpInit() {
 		strings.Split(config.GetKeyBinding(config.CmdLink0), ",")[0])
 
 	helpCells = fmt.Sprintf(helpCells,
+		config.GetKeyBinding(config.CmdMoveup),
+		config.GetKeyBinding(config.CmdMovedn),
+		config.GetKeyBinding(config.CmdMoveLeft),
+		config.GetKeyBinding(config.CmdMoveRight),
```
Comment (makeworld-the-better-one) [inline display/help.go:84]: You need to add a place for these arguments to go in the `helpCells` text.

File: display/tab.go [hunk 2]
```diff
@@ -128,7 +128,8 @@ func makeNewTab() *tab {
 
 		key := event.Key()
 		mod := event.Modifiers()
-		ru := event.Rune()
+		// ru := event.Rune()
+		cmd := config.TranslateKeyEvent(event)
```
Comment (makeworld-the-better-one) [inline display/tab.go:132]: Since the whole `event` is being looked at, get rid of the `key` and `mod` variables, as well as the `ru` comment. Then fix the logic below to just look at `cmd`.

File: display/tab.go [hunk 3]
```diff
@@ -158,26 +159,30 @@ func makeNewTab() *tab {
 				}
 			}
 			t.page.Column++
+			t.view.ScrollTo(t.page.Row, t.page.Column)
```
Comment (makeworld-the-better-one) [inline display/tab.go:162]: None of these added lines should be needed, please test without them. The actual scroll of the `TextView` is changed in `applyHorizontalScroll` and `applyScroll`.

**Review by makeworld-the-better-one — CHANGES_REQUESTED** [review]
Thanks for creating this PR! Please see my comments.

- Add defaults to `config/config.go`
- Test with pages that fit entirely on the screen, as well as pages that are too long, too wide, and both. Make sure scrolling works as expected. Keep in mind that the current scrolling code in master (not in v1.8.0) is slightly broken, see [this comment](https://github.com/makeworld-the-better-one/amfora/issues/197#issuecomment-812582553). If this code reveals a new error let me know, and we can figure out if it's this PR or in master.

**Comment by makeworld-the-better-one** [comment]
Note to self: update wiki keybindings once this is merged

File: config/default.go [hunk 4]
```diff
@@ -148,6 +148,10 @@ scrollbar = "auto"
 # bind_reload
 # bind_back
 # bind_forward
+# bind_moveup
+# bind_movedn
+# bind_moveleft
+# bind_moveright
```
Comment (makeworld-the-better-one) [inline config/default.go:154]: Remove these from this file. Add them to `default-config.toml` instead, and then run `go generate ./...`. Sorry this is not documented.

File: display/tab.go (hunk 3 above)
Comment (singalhimanshu) [reply display/tab.go:162]: why aren't we using `applyScroll` instead of `applyHorizontalScroll` here https://github.com/makeworld-the-better-one/amfora/blob/d4c7e87838d2ad2cc58d3b56995ee456a614e238/display/tab.go#L188

Comment (makeworld-the-better-one) [reply display/tab.go:162]: See the comments at the beginning of the section: https://github.com/makeworld-the-better-one/amfora/blob/d4c7e87838d2ad2cc58d3b56995ee456a614e238/display/tab.go#L126-L127

Whenever vertical scrolling happens, the keys are returned and the actual scrolling is handled by cview. If that needs to be changed it can, but it shouldn't matter for this PR.

Comment (singalhimanshu) [reply display/tab.go:162]: But this will cause issues when the user will change the `moveup` and `movedown` key-binds to something other than `k` and `j`. In that case cview won't handle the scrolling, it should be done manually by calling `applyScroll` and increment/decrement the value of `t.Page.Row`
One more issue is that `t.Page.Row` starts from 0 which means that even if we reach at the end of the page the value of `t.page.Row` can still increment.
This can be resolved by using `termH` and doing some mathematical operations to make sure that we don't overflow the value of `t.page.Row` beyond the `height`.

File: config/default.go (hunk 4 above)
Comment (singalhimanshu) [reply config/default.go:154]: I have added that to the `default-config.toml` `go generate ./...` generates this default.go so I guess no changes are required here.

File: display/tab.go (hunk 3 above)
Comment (makeworld-the-better-one) [reply display/tab.go:158]: Yes, you're correct. Those changes all sound good to me. That should be implemented in this PR. It means the `return`s will have to changed to `return nil` instead of `return event`, so the `TextView` (`t.view`) doesn't see the up and down keys.

File: config/default.go (hunk 4 above)
Comment (makeworld-the-better-one) [reply config/default.go:154]: Yep, seems good then.

**Comment by makeworld-the-better-one** [comment]
See my comments and let me know once you think all the required changes have been made. Thanks!

File: display/tab.go (hunk 3 above)
Comment (singalhimanshu) [reply display/tab.go:158]: @makeworld-the-better-one There's one more problem, the help page doesn't implement any custom scrolling it depends upon the scrolling from`cview`. Should I implement that as well in this PR?

Comment (singalhimanshu) [reply display/tab.go:158]: @makeworld-the-better-one any update on what should I do about the help page?

Comment (makeworld-the-better-one) [reply display/tab.go:158]: > But this will cause issues when the user will change the `moveup` and `movedown` key-binds to something other than `k` and `j`. In that case cview won't handle the scrolling, it should be done manually

Ok I actually figured out a solution to this, and it applies to help page too. cview has defaults for scrolling keys, but these can be overridden. See [here](https://docs.rocketnine.space/code.rocketnine.space/tslocum/cview#pkg-variables) and scroll down to `var Keys =`.

So basically what should be done is handle left/right scrolling keys here. For up/down scrolling, set the keys in the `cview.Keys` map before starting, like in the `config/` package. You should also set the left/right scroll keys, because despite them being captured here, they're needed for the help page as you pointed out.

Does that make sense? Thanks for waiting.

Comment (singalhimanshu) [reply display/tab.go]: Seems like a good solution. I have implemented this you can review that and let me know if they are in correct place or not.

Comment (makeworld-the-better-one) [reply display/tab.go]: Looks good!

**Comment by makeworld-the-better-one** [comment]
Thanks for updating this! Merged. I made some changes of my own at the end there. The only important one is dd83115. I added back the up/down scrolling lines and changed some of it to use the commands. I also track arrow keys for all scrolling in this section. That's to match the scrolling behaviour in the master branch, which requires tracking the scrolling in all directions, no matter what key is being used. You can see #197 for more details about this scroll behaviour.
